Srclib.Nuget/Graph/GraphRunner.cs

```diff
@@ -14,185 +14,297 @@
 using System.Xml.Linq;
 using Srclib.Nuget.Documentation;
 
+using Newtonsoft.Json;
+
 namespace Srclib.Nuget.Graph
 {
-  /// <summary>
-  /// C# syntax walker that produces graph output.
-  /// </summary>
-  public class GraphRunner : CSharpSyntaxWalker
-  {
-    readonly Output _output = new Output();
-    readonly List<Tuple<SyntaxToken, ISymbol, string>> _refs = new List<Tuple<SyntaxToken, ISymbol, string>>();
-    readonly HashSet<ISymbol> _defined = new HashSet<ISymbol>();
-    SemanticModel _sm;
-    string _path;
-
-    private GraphRunner()
-      : base(SyntaxWalkerDepth.Token)
-    {
-    }
-
     /// <summary>
-    /// Add a definition to the output.
-    /// Also adds a ref to the def, and
-    /// potentially a doc.
+    /// C# syntax walker that produces graph output.
     /// </summary>
-    /// <param name="def">The def to add.</param>
-    /// <param name="symbol">The symbol the def was created from.</param>
-    void AddDef(Def def, Doc doc = null)
+    public class GraphRunner : CSharpSyntaxWalker
     {
-      var r = Ref.AtDef(def);
-      _output.Defs.Add(def);
-      _output.Refs.Add(r);
-
-      if (doc != null)
-      {
-        doc.UnitType = r.DefUnitType;
-        doc.Unit = r.DefUnit;
-        doc.Path = r.DefPath;
-        _output.Docs.Add(doc);
-      }
-    }
+        readonly Output _output = new Output();
+        readonly List<Tuple<SyntaxToken, ISymbol, string>> _refs = new List<Tuple<SyntaxToken, ISymbol, string>>();
+        readonly HashSet<ISymbol> _defined = new HashSet<ISymbol>();
+        readonly HashSet<string> keys = new HashSet<string>();
+        readonly static Dictionary<string, string> dllToProjectUrl = new Dictionary<string, string>();
+        readonly static Dictionary<string, string> projectUrlToRepo = new Dictionary<string, string>();
 
-    /// <summary>
-    /// Scan a collection of resolved tokens and generate a ref for all 
-    /// tokens that have a corresponding def 
-    /// </summary>
-    void RunTokens()
-    {
-      foreach(var r in _refs)
-      {
-        var token = r.Item1;
-        var definition = r.Item2;
-        var file = r.Item3;
+        SemanticModel _sm;
+        string _path;
+
+        private GraphRunner() : base(SyntaxWalkerDepth.Token)
+        {
+            projectUrlToRepo["http://www.newtonsoft.com/json"] = "github.com/JamesNK/Newtonsoft.Json";
+            projectUrlToRepo["http://autofac.org/"] = "github.com/autofac/Autofac";
+            projectUrlToRepo["http://msdn.com/roslyn"] = "github.com/dotnet/roslyn";
+            projectUrlToRepo["http://www.asp.net/"] = "github.com/aspnet/dnx";//this may not be always correct
+        }
 
-        if (_defined.Contains(definition))
+        /// <summary>
```

Comment:
Let's follow the MSDN C# style guidelines when it comes to comments and everything else in the code: https://msdn.microsoft.com/en-us/library/ff926074.aspx